libexec/tfenv-install

```diff
@@ -51,13 +51,14 @@ fi
 
 case "$(uname -s)" in
   Darwin*)
-    os="darwin_amd64"
+    os="darwin_${TFENV_ARCH}"
     ;;
   MINGW64*)
-    os="windows_amd64"
+    os="windows_${TFENV_ARCH}"
     ;;
   *)
-    os="linux_amd64"
+    TFENV_ARCH=${TFENV_ARCH:-amd64}
```

Comment:
Please move this line before the `case` statement (L51) because `os` ends up invalid due to `TFENV_ARCH` is empty when OS is not Linux.